src/ring_theory/finiteness.lean

```diff
@@ -564,7 +564,45 @@ section monoid_algebra
 
 namespace add_monoid_algebra
 
-open algebra add_submonoid
+open algebra add_submonoid submodule
+
+section span
+
+variables {R : Type*} {M : Type*} [comm_semiring R] [add_monoid M]
+
+lemma mem_adjoint_support (f : (add_monoid_algebra R M)) :
+  f ∈ adjoin R ((add_monoid_algebra.of R M ∘ multiplicative.of_add)'' (f.support : set M)) :=
+begin
+  suffices : span R (of R M '' (f.support : set M)) ≤
+    (adjoin R (of R M '' (f.support : set M))).to_submodule,
+  { exact this (mem_span_support f) },
+  rw submodule.span_le,
+  exact subset_adjoin
+end
+
+lemma support_gen_of_gen {S : set (add_monoid_algebra R M)} (hS : algebra.adjoin R S = ⊤) :
+  algebra.adjoin R (⋃ f ∈ S, ((of R M)'' (f.support : set M))) = ⊤ :=
```

Comment:
Again, this is making a mess of the types
```suggestion
  algebra.adjoin R (⋃ f ∈ S, ((of R M ∘ multiplicative.of_add) '' (f.support : set M))) = ⊤ :=
```
Frankly it might be worth defining an `of'` that doesn't expect multiplicative in the first place.

Reply:
I can do it, but it cannot be a bundled hom, right? Since the domain is additive, but addition becomes multiplication in the algebra.

Reply:
Right, but you're not using the bundling anyway here